pygeos/_geometry.pyx

```diff
@@ -134,37 +136,43 @@ def simple_geometries_1d(object coordinates, object indices, int geometry_type,
-                            break
-                # check the resulting size to prevent invalid rings
-                if geom_size + ring_closure < 4:
-                    # the error equals PGERR_LINEARRING_NCOORDS (in pygeos/src/geos.h)
-                    raise ValueError("A linearring requires at least 4 coordinates.")
-
-            seq = GEOSCoordSeq_create_r(geos_handle, geom_size + ring_closure, dims)
-            for coord_idx in range(geom_size):
-                if _set_xyz(geos_handle, seq, coord_idx, dims, coord_view, idx) == 0:
-                    GEOSCoordSeq_destroy_r(geos_handle, seq)
-                    return  # GEOSException is raised by get_geos_handle
-                idx += 1
+            # # check if we need to close a linearring
+            # if geometry_type == 2:
+            #     ring_closure = 0
+            #     if geom_size == 3:
+            #         ring_closure = 1
+            #     else:
+            #         for coord_idx in range(dims):
+            #             if coord_view[idx, coord_idx] != coord_view[idx + geom_size - 1, coord_idx]:
+            #                 ring_closure = 1
+            #                 break
+            #     # check the resulting size to prevent invalid rings
+            #     if geom_size + ring_closure < 4:
+            #         # the error equals PGERR_LINEARRING_NCOORDS (in pygeos/src/geos.h)
+            #         raise ValueError("A linearring requires at least 4 coordinates.")
+
+            # seq = GEOSCoordSeq_create_r(geos_handle, geom_size + ring_closure, dims)
+            # for coord_idx in range(geom_size):
+            #     if _set_xyz(geos_handle, seq, coord_idx, dims, coord_view, idx) == 0:
+            #         GEOSCoordSeq_destroy_r(geos_handle, seq)
+            #         return  # GEOSException is raised by get_geos_handle
+            #     idx += 1
+
+            seq = GEOSCoordSeq_copyFromBuffer_r(geos_handle, &coord_view[idx*dims], geom_size, has_z, 0)
+
+            if seq == NULL:
+                return  # GEOSException is raised by get_geos_handle
+            idx += geom_size
 
             if geometry_type == 0:
                 geom = GEOSGeom_createPoint_r(geos_handle, seq)
             elif geometry_type == 1:
                 geom = GEOSGeom_createLineString_r(geos_handle, seq)
             elif geometry_type == 2:
-                if ring_closure == 1:
-                    if _set_xyz(geos_handle, seq, geom_size, dims, coord_view, idx - geom_size) == 0:
-                        GEOSCoordSeq_destroy_r(geos_handle, seq)
-                        return  # GEOSException is raised by get_geos_handle
+                # if ring_closure == 1:
+                #     if _set_xyz(geos_handle, seq, geom_size, dims, coord_view, idx - geom_size) == 0:
+                #         GEOSCoordSeq_destroy_r(geos_handle, seq)
+                #         return  # GEOSException is raised by get_geos_handle
```

Comment:
When using the `GEOSCoordSeq_copyFromBuffer_r`, I am note sure we should automatically add a coordinate for ring closure.

(although I suppose it is still possible; we can first copy in one extra element (the first of the next geometry?) and then overwrite that later?)

Personally I think that when you already are providing `indices`, it should also be a rare use case (eg if you got those indices from one of our functions that returns indices, this will never be a problem). 
We could of course still check for it (now I also commented out that code). 

This could maybe also be an option of the function?